src/dash-playlist-loader.js

```diff
@@ -277,20 +282,15 @@ export default class DashPlaylistLoader extends EventTarget {
 
     // initialize the loader state
     // The masterPlaylistLoader will be created with a string
-    if (typeof srcUrlOrPlaylist === 'string') {
+    if (this.isMaster_) {
```

Comment:
should we verify that `srcUrlOrPlaylist` is a string or is it basically guaranteed to be one if we're a master manifest?

Reply:
For master it will always be a string, for others it can be a playlist or string. So yes it would seem so.